### PR TITLE
Fix Unhandled DBErrors in ComponentsWebAssembly and Spa templates

### PR DESCRIPTION
**Pull request by JunTaoLuo**
Fixes https://github.com/dotnet/aspnetcore/issues/25348. The original change had a bug where calls to AddDatabaseDeveloperPageExceptionFilter were nested in a #else clause by mistake.

I'll verify these changes manually.

### CODE REVIEW

**Review by pranavkm — APPROVED** [review]

**Comment by JunTaoLuo** [comment]
During verification I found another issue that caused a null ref: https://github.com/dotnet/efcore/issues/22341. Turns out @javiercn also saw the same problem during his attempt to resolve the issue. I've added a workaround for the problem and was able to manually verify that the issue has been fixed.

**Comment by JunTaoLuo** [comment]
FYI @Pilchie

**Comment by Pilchie** [comment]
@pranavkm - can you take another look with the changes?

Approved for RC1 pending CI completion, and @pranavkm's signoff if merged before 10am Pacific on 2020-09-01.

File: src/Middleware/Diagnostics.EntityFrameworkCore/src/DatabaseDeveloperPageExceptionFilter.cs
Comment (pranavkm) [inline src/Middleware/Diagnostics.EntityFrameworkCore/src/DatabaseDeveloperPageExceptionFilter.cs:42]: Since the PR is still building, could we make a small note about the workaround:

```suggestion
                    .Distinct(); // Workaround for https://github.com/dotnet/efcore/issues/22341
```

**Review by pranavkm — APPROVED** [review]

**Review by javiercn — APPROVED** [review]

**Pull request merged**
